adapters/adapterstest/test_json.go

```diff
@@ -101,24 +101,28 @@ func runSpec(t *testing.T, filename string, spec *testSpec, bidder adapters.Bidd
 	diffErrorLists(t, fmt.Sprintf("%s: MakeRequests", filename), errs, spec.MakeRequestErrors)
 	diffHttpRequestLists(t, filename, actualReqs, spec.HttpCalls)
 
-	var bids = make([]*adapters.TypedBid, 0, len(spec.Bids))
+	bidResponse := adapters.NewBidderResponse()
+
 	var bidsErrs = make([]error, 0, len(spec.MakeBidsErrors))
 	for i := 0; i < len(actualReqs); i++ {
-		theseBids, theseErrs := bidder.MakeBids(&spec.BidRequest, spec.HttpCalls[i].Request.ToRequestData(t), spec.HttpCalls[i].Response.ToResponseData(t))
-		bids = append(bids, theseBids...)
+		thisBidResponse, theseErrs := bidder.MakeBids(&spec.BidRequest, spec.HttpCalls[i].Request.ToRequestData(t), spec.HttpCalls[i].Response.ToResponseData(t))
 		bidsErrs = append(bidsErrs, theseErrs...)
+		if thisBidResponse != nil {
+			// The currency is the same accross all bids
```

Comment:
Fair point, I agree with you, I make sure that currency is populated everytime:
      if thisBidResponse != nil {
           bidResponse.Currency = thisBidResponse.Currency
           bidResponse.Bids = append(bidResponse.Bids, thisBidResponse.Bids...)
       }

Reply:
wait... I'm a little bit confused by what this accomplishes.

I might be missing something... but it looks like the currency from the _last_ call to `MakeBids` takes effect now, rather than the first one.

Doesn't seem like it addresses the issues with non-determinism from parallel calls?

Reply:
Oh sorry, I get what you meant, just updated the change, let me know what you think.

Reply:
perfect :). Thanks!